UITests/XCUIElement+Utils.swift

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2017-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import XCTest
+
+extension XCUIElement {
+    
+    var stringValue: String? {
+        value as? String
+    }
+    
+    func waitForText(predicate: @escaping (String?) -> Bool, timeout: TimeInterval) -> Bool {
+        var result = false
+        var secondsToInvalidate = timeout
+        
+        let semaphore = DispatchSemaphore(value: 1)
+        
+        while !secondsToInvalidate.isLessThanOrEqualTo(.zero) && !result {
+            if predicate(self.stringValue) {
+                result = true
+                semaphore.signal()
+                break
+            }
+            
+            secondsToInvalidate -= 1
+            _ = semaphore.wait(timeout: .now() + 1)
+        }
+        
+        return result
+    }
+    
+    func clearText(app: XCUIApplication? = nil) {
```

Comment:
I get why you're doing this...and it _is_ very frustrating dealing with text input views in XCUI tests after iOS 10.x. It's sad that we have to resort to these workarounds, and sometimes I find it easier to simply add toolbar buttons or something similar that lets us reset this from within the app itself.

For example, we could have a toolbar on all UIViewControllers that resets all the state / text fields within it, which the automated tests could trigger much simpler.

Reply:
Agree with you, but we don't control JS sign-in widget' text fields, it comes with own toolbar, own return key and own style :(